### ISSUE TITLE
Fix typing inconsistencies

### ISSUE THREAD

**Issue by vcianciarelli** [Adyen/adyen-web#740]
It's not a bug. This is related to incorrect typings that I noticed.

I found them in two places:

1. https://github.com/Adyen/adyen-web/blob/1f879c4f27439d9904638ccc12db0606dd433487/packages/lib/src/components/ApplePay/types.ts#L21

I've verified that it could be a number based on the following:

a) https://github.com/Adyen/adyen-web/blob/1f879c4f27439d9904638ccc12db0606dd433487/packages/lib/src/components/ApplePay/utils.ts#L8-L14

and b) Adyen's documentation on how to implement Apple Pay in drop-in [is showed as such](https://docs.adyen.com/payment-methods/apple-pay/web-drop-in#configuration):

<img width="814" alt="Screen Shot 2021-02-10 at 9 57 38 AM" src="https://user-images.githubusercontent.com/37092143/107527051-7654f100-6b86-11eb-8370-a4dcef7ca3cb.png">

The solution would be to type it as `PaymentAmount | number`.

2. https://github.com/Adyen/adyen-web/blob/1f879c4f27439d9904638ccc12db0606dd433487/packages/lib/src/components/ApplePay/types.ts#L117

`onAuthorized` can accept another parameter, which you can see in this [link](https://developer.apple.com/documentation/apple_pay_on_the_web/applepaysession/1778020-onpaymentauthorized). In our code, we typed this method as

```
onAuthorized: (resolve, reject, data) => void
```

The reason why I think these typings should be fixed is because I get Typescript complaining that I'm not using `AdyenCheckout` library properly, which doesn't seem to be the case.

**Comment by marcperez** [comment]
Hi @vcianciarelli,

Thanks for pointing this out, it appears that Apple Pay code sample in our docs is using our old amount format, we'll make sure to update it. The correct way of setting the amount is as the types indicate (although we've been temporarily accepting both formats). For example:

```js
{
  // ...
  amount: {
    value: 1000,
    currency: "USD"
  }
  // ...
}
```

On the missing parameter in the `onAuthorized` event type, you're absolutely right, we'll update this issue as soon as we have a fix.

Hope this helps!

**Comment by marcperez** [comment]
Hello again @vcianciarelli,

The fix on the Apple Pay types has been merged (as part of #751) and it will be released in `v4.0.0`. Thanks again for your feedback!
